sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1294,20 +1296,29 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
       userSpecifiedColumns: Seq[(String, Option[String])],
       query: QueryContext,
       properties: Map[String, String],
-      allowExisting: Boolean,
+      ignoreIfExists: Boolean,
       replace: Boolean,
-      isTemporary: Boolean): LogicalPlan = {
+      isTemporary: Boolean,
+      isAlterViewAsSelect: Boolean): LogicalPlan = {
```

Comment:
how about a `viewType` parameter which is an enum?
```
public enum SaveMode {
  Temporary,
  Permanent,
  Either
}
```
We are going to add `global temp view` soon, so this enum will also be useful at that time.